lib/ProductOpener/Products.pm

```diff
@@ -2954,6 +2955,32 @@ or "slack_CHANNEL_NAME" (B<warning> currently channel name is ignored, we post t
 
 =cut
 
+sub removeEmailValues ($product_ref) {
+
+	# Iterate over the product fields
+	foreach my $field (keys %{$product_ref->{product}}) {
+
+		if (defined $product_ref->{product}->{$field}) {
+			my $value = $product_ref->{product}->{$field};
+			my $emailRegex = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+
+			# Check if the field is multi-valued
+			if (ref($value) eq 'ARRAY') {
+				my @new_values;
+				foreach my $entry (@$value) {
+					push @new_values, $entry unless $entry && $entry =~ $emailRegex;
```

Comment:
A product can have nested arrays / hashes. For instance the "packagings" field is an array of "packaging components" hashes. So this could apply the regexp match to a reference to an array or hash, which would not work.

Reply:
You are right! It needs to be modified to handle references to arrays or hashes correctly.
Thanks for pointing out!